pkg/ctl/mcp/tools.go

```diff
@@ -0,0 +1,323 @@
+// Package mcp implements the Model Context Protocol (MCP) server functionality for eksctl
+package mcp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+const (
+	CommandExecutionTimeout = 45 * time.Second
+	HandlerTimeout          = 5 * time.Minute
+)
+
+// registerTools registers all eksctl commands as MCP tools
+// This is the entry point for tool registration that processes the entire command tree
+func registerTools(s *server.MCPServer, rootCmd *cobra.Command) error {
+	return registerToolsRecursive(s, rootCmd)
+}
+
+// registerToolsRecursive recursively registers all commands as tools
+// It processes the current command and then recursively processes all subcommands
+func registerToolsRecursive(s *server.MCPServer, cmd *cobra.Command) error {
+	if err := registerTool(s, cmd); err != nil {
+		return err
+	}
+
+	// Process all subcommands recursively
+	for _, subCmd := range cmd.Commands() {
+		if err := registerToolsRecursive(s, subCmd); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// registerTool registers a single command as an MCP tool
+// It builds the tool options and creates a handler for the command
+func registerTool(s *server.MCPServer, cmd *cobra.Command) error {
+	// Skip commands that shouldn't be exposed
+	if shouldSkipCommand(cmd) {
+		return nil
+	}
+
+	// Build tool name and options
+	toolName, toolOptions, err := buildToolOptions(cmd)
+	if err != nil {
+		return fmt.Errorf("error building tool options for %s: %w", cmd.Name(), err)
+	}
+
+	if toolName == "" {
+		return nil
+	}
+
+	// Create and register the tool
+	tool := mcp.NewTool(toolName, toolOptions...)
+	s.AddTool(tool, createToolHandler(cmd))
+
+	return nil
+}
+
+// shouldSkipCommand determines if a command should be skipped for MCP tool registration
+func shouldSkipCommand(cmd *cobra.Command) bool {
+	return cmd.Hidden || cmd.Name() == "help" || cmd.Name() == "completion"
+}
+
+// buildToolOptions builds the MCP tool options from a cobra command
```

Comment:
Nice to see the MCP tools' schemas and descriptions getting generated from the code in this automated way!